HDDS-11938. Add overview/docs landing page (#136)
apache/ozone-site: docs/01-overview.md

- **Billions of Objects**: Designed from the ground up to store and manage billions of objects efficiently
- **Separation of Namespaces**: Decouples namespace management from block space management, allowing independent scaling on both axes
- **Dense Storage Support**: Optimized for high-density storage nodes with support for up to 400TB per node (compared to 100TB in traditional HDFS)

### Multi-Protocol Support

- **S3 Compatible API**: Native support for the Amazon S3 API, enabling seamless integration with S3-compatible tools and applications
- **Hadoop Compatible File System**: Provides a filesystem interface that works with existing Hadoop ecosystem applications
- **Rich API Options**: Supports multiple client interfaces including Java API, command-line tools, and REST endpoints

### Enterprise-Ready Security

- **Strong Authentication**: Integrated Kerberos authentication with robust security mechanisms
- **Fine-Grained Authorization**: Support for both native ACLs and Apache Ranger integration for centralized authorization policies
- **Encryption**: Transparent data encryption at rest and in-flight to protect sensitive information

### Robust Data Management

> - **Strong Consistency**: Provides strict consistency to simplify application design and ensure data integrity

Reviewer: Strong or strict? I think there's a technical difference but I don't recall exactly.

- **Metadata Management**: Efficient handling of metadata with dedicated services for high performance
- **Snapshots**: Support for point-in-time snapshots to protect against data corruption and facilitate backups

> ### Operational Excellence

Reviewer: This section is pretty random, although the individual points are valid. Can we regroup them elsewhere? EC/replication and fault tolerance probably belong under "Robust Data Management". Observability could be its own section.

- **Replication and Erasure Coding**: Flexible data protection strategies to balance storage efficiency and durability requirements

### Integration with Data Analytics Ecosystem

- **Hadoop Ecosystem**: Seamless integration with Hadoop, Hive, and Spark workloads
- **SQL Engines**: Works with SQL query engines like Hive, Impala, and Trino without modification
- **Modern Data Formats**: Supports modern table formats like Apache Iceberg for data lake architectures

## Architecture Overview

Ozone has a layered architecture that separates namespace management from storage management:

- **Ozone Manager (OM)**: Manages the namespace hierarchy (volumes, buckets, and keys) and handles client metadata operations
- **Storage Container Manager (SCM)**: Manages the containers where data is stored and handles block allocation
- **Datanodes**: Store the actual data in containers and provide read and write access
- **Recon**: Analytics and monitoring service that provides insight into the cluster

This separation allows Ozone to achieve the scale required for modern storage systems while maintaining high performance and reliability.

## Storage Elements

> Ozone organizes storage in a three-level hierarchy:

Reviewer: We should add a link to the namespace overview page somewhere in this section for more information too.

- **Volumes**: Similar to accounts, created by administrators for organizations or teams
- **Buckets**: Created by users within volumes, similar to S3 buckets
- **Keys**: Data objects stored inside buckets, each potentially containing multiple blocks

> When a client writes data, Ozone stores it on Datanodes in chunks called blocks, which are organized into containers for efficient management and replication.

Reviewer: This seems better suited for the architecture section. You can use Ozone's namespace layout without knowing this implementation detail.
```suggestion
When a client writes data, Ozone stores it as blocks on the Datanodes, which are organized into storage containers for efficient management and replication.
```

## Getting Started

> To get started with Ozone, see the [Quick Start Guide](./02-quick-start/01-installation/01-docker.md) for installation instructions and basic usage examples.

Reviewer: Update this PR with the latest feature branch to get a landing page for each subsection. Also do not include the numbered prefixes in links since they'll break (and fail the build) if sections are changed or moved. Docusaurus will resolve the links without them.
```suggestion
To get started with Ozone, see the [Quick Start Guide](quick-start) for installation instructions and basic usage examples.
```
